src/mlpack/methods/matrix_completion/matrix_completion.cpp

```diff
@@ -59,13 +60,9 @@ void MatrixCompletion::CheckValues()
         << "indices does not have 2 rows!" << std::endl;
   }
 
-  if (indices.n_cols != values.n_elem)
-  {
-    Log::Fatal << "MatrixCompletion::CheckValues(): the number of constraint "
-        << "indices (columns of constraint indices matrix) does not match the "
-        << "number of constraint values (length of constraint value vector)!"
-        << std::endl;
-  }
+  arma::mat transposeValues = values.t();
```

Comment:
This will actually compute the transpose of values---it would be better to avoid this because that may be computationally expensive.  Perhaps you can modify `util::CheckSameSize()` to have an option that one of its arguments should be transposed?